datafusion/sql/src/expr/value.rs

```diff
@@ -394,6 +386,23 @@ const fn try_decode_hex_char(c: u8) -> Option<u8> {
     }
 }
 
+fn parse_decimal_128_without_scale(s: &str) -> Result<i128> {
+    let number = s.parse::<i128>().map_err(|e| {
+        DataFusionError::from(ParserError(format!(
+            "Cannot parse {s} as i128 when building decimal: {e}"
+        )))
+    })?;
+
+    const MAX_DECIMAL_128_VALUE: i128 = 10_i128.pow(38) - 1;
```

Comment:
Perhaps using https://docs.rs/arrow/latest/arrow/datatypes/constant.DECIMAL128_MAX_PRECISION.html would be clearer than `38` here

Reply:
Fixed. Thank you @alamb .